app/controllers/billing_invoice.py

```diff
@@ -118,3 +121,64 @@ def post(self, project_id):
             message=f'Invoice for user with email {user.email} and project_id {project_id} sent successfully',
             data=dict(invoice=new_invoice_data)
         ), 200
+
+
+    @jwt_required
+    def get(self, project_id):
+
+        current_user_id = get_jwt_identity()
+        current_user_roles = get_jwt_claims()['roles']
+
+        billing_invoice_schema = BillingInvoiceSchema(many=True)
+        project = Project.get_by_id(project_id)
+
+        billing_invoice = BillingInvoice.find_all(project_id=project_id)
+
+        if not billing_invoice:
+            return dict(
+                status='fail',
+                message=f'billing invoice records not found'
+            ), 404
+        
+        
+        if not is_owner_or_admin(project, current_user_id, current_user_roles):
+                return dict(status='fail', message='Unauthorised'), 403
+
+        billing_invoice_data, errors = billing_invoice_schema.dumps(billing_invoice)
+
+        if errors:
+            return dict(status='fail', message=errors), 500
+
+        return dict(status='success', data=dict(
+            transaction=json.loads(billing_invoice_data))), 200
+
+
+class BillingInvoiceDetailView(Resource):
+    
+    @jwt_required
+    def get(self,project_id ,invoice_id):
+
+        current_user_id = get_jwt_identity()
+        current_user_roles = get_jwt_claims()['roles']
+
+        transaction_schema = BillingInvoiceSchema()
+        project = Project.get_by_id(project_id)
+
+        billing_invoice = BillingInvoice.get_by_id(invoice_id)
+
+        if not billing_invoice:
+            return dict(
+                status='fail',
+                message=f'billing invoice with invoice id {invoice_id} not found'
+            ), 404
+
+        if not is_owner_or_admin(project, current_user_id, current_user_roles):
+                return dict(status='fail', message='Unauthorised'), 403
+
+        billing_invoice_data, errors = transaction_schema.dumps(billing_invoice)
+        
+        if errors:
+            return dict(status='fail', message=errors), 500
+
+        return dict(status='success', data=dict(
+            transaction=json.loads(billing_invoice_data))), 200
```

Comment:
Since this is not a transaction route, I believe it should be better to use `invoice`
```suggestion
            invoice=json.loads(billing_invoice_data))), 200
```